src/proto/uprotocol/uauthority.rs

```diff
@@ -11,16 +11,104 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+use bytes::BufMut;
+
 use crate::uprotocol::uri::uauthority::Number;
 use crate::uprotocol::UAuthority;
 
-pub use crate::uri::validator::ValidationError;
+pub use crate::proto::uprotocol::uuri::SerializationError;
+use crate::uri::validator::ValidationError;
 
 const REMOTE_IPV4_BYTES: usize = 4;
 const REMOTE_IPV6_BYTES: usize = 16;
 const REMOTE_ID_MINIMUM_BYTES: usize = 1;
 const REMOTE_ID_MAXIMUM_BYTES: usize = 255;
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum AddressType {
```

Comment:
FMOPV this function needs some doc comments which provide the link to the specification of the micro URI serialization format.

In the code it would be helpful to add a comment explaining that the assigned integers actually correspond to the values used in the micro URI format's *TYPE* field ...

Reply:
Done